qiskit_experiments/library/randomized_benchmarking/rb_experiment.py

```diff
@@ -192,80 +276,125 @@ def _sequences_to_circuits(
-            if self.num_qubits == 2:
-                return CliffordUtils.clifford_2_qubit(op).adjoint()
+        if self.num_qubits == 1:
+            return inverse_1q(op)
+        if self.num_qubits == 2:
+            return inverse_2q(op)
+        if isinstance(op, QuantumCircuit):
+            return Clifford.from_circuit(op).adjoint()
         return op.adjoint()
 
     def _transpiled_circuits(self) -> List[QuantumCircuit]:
         """Return a list of experiment circuits, transpiled."""
-        # TODO: Custom transpilation (without calling transpile()) for 1Q and 2Q cases
-        transpiled = super()._transpiled_circuits()
+        has_custom_transpile_option = (
+            any(
+                opt not in {"basis_gates", "optimization_level"}
+                for opt in vars(self.transpile_options)
+            )
+            or self.transpile_options.get("optimization_level", 0) != 0
+        )
+        has_no_undirected_2q_basis = self._get_basis_gates() is None
+        if self.num_qubits > 2 or has_custom_transpile_option or has_no_undirected_2q_basis:
+            transpiled = super()._transpiled_circuits()
+        else:
+            transpiled = [
+                _transpile_clifford_circuit(circ, physical_qubits=self.physical_qubits)
+                for circ in self.circuits()
+            ]
+            # Set custom calibrations provided in backend
+            if isinstance(self.backend, BackendV2):
+                # assert self.num_qubits <= 2
+                qargs_patterns = [self.physical_qubits]  # for self.num_qubits == 1
+                if self.num_qubits == 2:
+                    qargs_patterns = [
+                        (self.physical_qubits[0],),
+                        (self.physical_qubits[1],),
+                        self.physical_qubits,
+                        (self.physical_qubits[1], self.physical_qubits[0]),
+                    ]
+
+                instructions = []  # (op_name, qargs) for each element where qargs means qubit tuple
+                for qargs in qargs_patterns:
+                    for op_name in self.backend.target.operation_names_for_qargs(qargs):
+                        instructions.append((op_name, qargs))
+
+                common_calibrations = defaultdict(dict)
+                for op_name, qargs in instructions:
+                    inst_prop = self.backend.target[op_name].get(qargs, None)
+                    if inst_prop is None:
+                        continue
+                    schedule = inst_prop.calibration
+                    if schedule is None:
+                        continue
+                    publisher = schedule.metadata.get("publisher", CalibrationPublisher.QISKIT)
+                    if publisher != CalibrationPublisher.BACKEND_PROVIDER:
+                        common_calibrations[op_name][(qargs, tuple())] = schedule
+
+                for circ in transpiled:
+                    circ.calibrations = common_calibrations
```

Comment:
Actually, it's not a new behavior introduced by this PR. Previously, the setting of circ.calibrations was done implicitly in [PulseGate](https://qiskit.org/documentation/stubs/qiskit.transpiler.passes.PulseGates.html) pass run in   `transpile` called in `_transpile_circuits`. So I might include the logic in `_transpile_clifford_circuit` but I didn't. I just thought creating `common_calibrations` once outside of it would be faster than creating different calibration for each circuit.

Reply:
Thanks for the explanation.